File: Data Structures & Algorithms/is-anagram/submission-4.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;

        int[] alphabet = new int[26];

        for (int idx = 0; idx < s.length(); idx++){
            alphabet[s.charAt(idx) - 'a']++;
            alphabet[t.charAt(idx) - 'a']--;
        }

        for (int idx = 0; idx < 26; idx++){
            if (alphabet[idx] != 0) return false;
        }
        
        return true;

    }
}

File: Data Structures & Algorithms/anagram-groups/submission-4.cpp
using namespace std;
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> output;

        for (size_t idx = 0; idx < strs.size(); idx++){
            string key = strs[idx];
            sort(key.begin(), key.end());
            output[key].push_back(strs[idx]);
        }

        vector<vector<string>> result;
        for (auto& [key, val] : output) {
            result.push_back(val);
        }
        return result;
    
    }
};
